Build fails on Win-On-ARM when compiled with gcc/clang

# Issue description

```
6.703 /50-libzmq/src/clock.cpp:251:48: error: use of undeclared identifier 'nsecs_per_usec'; did you mean 'usecs_per_sec'?
6.703   251 |     return static_cast<uint64_t> (ts.tv_sec) * nsecs_per_usec * usecs_per_sec
6.705       |                                                ^~~~~~~~~~~~~~
6.705       [ 34%] Building CXX object CMakeFiles/libzmq-static.dir/src/options.cpp.obj
6.705 |                                                usecs_per_sec
6.705 /50-libzmq/src/clock.cpp:106:16: note: 'usecs_per_sec' declared here
6.705   106 | const uint64_t usecs_per_sec = 1000000;
6.705       |                ^
```

# Environment

* libzmq version (commit hash if unreleased): b95d94935ed107679fd0ad9efd2f3d47309b6fd3
* OS: Linux cross compiling for Win-on-Aarch64 via clang

# Minimal test code / Steps to reproduce the issue

1.  Compile libzmq for Win-On-AArch64 via clang


# What's the actual result? (include assertion message & call stack if applicable)

See error above

# What's the expected result?

The code for zmq::clock_t::rdtsc does not seem to have any provisions for that combination of platform and compiler.
It seems to need gcc/clang compatible asm instructions for at least aarch64.
Not sure if 32bit arm is something that matter for that platform.

--- a/src/clock.cpp
+++ b/src/clock.cpp
@@ -223,6 +223,10 @@ uint64_t zmq::clock_t::rdtsc ()
                                  ((13 & 15) << 3) | // crm
                                  ((0 & 7) << 0));   // op2
     return _ReadStatusReg (pmccntr_el0);
+#elif (defined(_WIN32) && defined(__GNUC__) && defined(__aarch64__))
+    uint64_t val;
+    __asm__ volatile("mrs %0, pmccntr_el0" : "=r"(val));
+    return val;
 #elif (defined __GNUC__ && (defined __i386__ || defined __x86_64__))
     uint32_t low, high;
     __asm__ volatile("rdtsc" : "=a"(low), "=d"(high));

